File: conjecture.py
#Rewrote my conjecture in python because it supports infinite precision
import time

def isprime(n):
    n = abs(int(n))
    if n < 2:
        return False
    if n == 2: 
        return True    
    if not n & 1: 
        return False
    for x in range(3, int(n**0.5)+1, 2):
        if n % x == 0:
            return False
    return True

for x in range(2,1001):
	counter = 0
	if isprime(x):
		num = x
		while num != 2:
			if isprime(num):
				num = (num*num)-1
			elif num % 2 == 0:
				num /= 2
			else:
				num += 1
			counter += 1
		print(x, " took", counter, "steps")

File: conjecture.cpp
//Calculates different paths to 2 of a conjecture I devised (Use python version as it supports infinite precision...)
#include <iostream>
#include <cmath>
#include <unistd.h>	
#include <cmath>

using namespace std;

bool isNumPrime(long num);

int main()
{
	bool status = false;
	bool firstRun = true;
	long num;
	long  totalCounter = 0, touchedPrimes = 0, touchedComposites = 0; 
	cout << "Enter number: ";
	cin >> num;
	long numCheck = num;
	do
	{
		cout << num;
		if(isNumPrime(num))
		{
			cout << " (prime)" << endl;
			num = (num*num)-1;
			touchedPrimes++;
		} else
		{
			cout << " (composite) " << endl;
			if(num % 2 == 0)
			{
				num/=2;
			} else 
			{
				num++;
			}
			touchedComposites++;
		}
		if(num == 2)
			status = true;
		if(numCheck == num && firstRun == false)
			status = true;
		usleep(50000);
		totalCounter++;
		firstRun = false;
	} while(status == false);
	cout << 2 << " (prime)" << endl;
	cout << "Took " << totalCounter << " steps" << "| Touched " << touchedPrimes << " primes" << " | Touched " << touchedComposites << " composites" << endl;
}

bool isNumPrime(long num) 
{
    if (num%2 == 0 && num != 2) 
    {
        return false;
    }

    for (long i = 3; i <= sqrt(num); i = i+2) 
    {
        if (num%i == 0) 
        {
            return false;
        }
    }
    return true;
}